### PR TITLE
Refactor documentation and test nomenclature

### PR DESCRIPTION
**Pull request by olivroy**
Sorry.. this got out of hand again. Please let me know if you agree with this. If not, I can send smaller separate PRs of the relevant bits!

TL;DR: The goal is basically to simplify the `x()` function to `x()` where applicable as recommended in the [documentation tidyverse style guide](https://style.tidyverse.org/documentation.html#r-code). (only the function bullet)

This PR started as a follow-up to #1683 as I wanted to make clarifications and improved some things I did there after taking a fresh look.
cbb1dca8179c371d260190568eac7d7add1e746e and 018b232b793e99f50856bf7232c80336fa0d1c8d do this. I finally discovered the way to avoid spurious ([snapshot](https://github.com/r-lib/testthat/issues/1958) git diffs while working on Windows: Set Posix line ending in Rproj (13b2666ad31fe955c132ac6574017d9b3cba45c6). Used https://www.tidyverse.org/blog/2020/03/roxygen2-7-1-0/ as ref too.

I'd like to get your views on how I edited the shiny related functions docs 64e80b348b61087a530ab3ecb37c997b93943293. (I removed details that felt redundant? Let me know if you agree. Otherwise, I will rework it!

Changes 
* Renamed `is_non_empty_string()` to `is_non_empty_chr()` for clarity (accepts length > 1) e0c4a10e11b4e4c7b77c7376ea304c286efb74c2 (string is usually for character + length = 1)
* renamed tests to use the convention ab0ed09f8142a9e11a675e1605a21da437f9a453

* Moved some test helper functions into own [test helper files](https://testthat.r-lib.org/articles/special-files.html#helper-files) to improve testing workflow (i.e. functions are loaded with `load_all()` af9cdd6
* Removed duplicate definition of test helper in tests/testthat/helper.R bcbb80dafedb023d53f249630328d0f3ee791f91
* Removed the check for xml2 installation since it is import now (I may do a follow-up where I remove `check_suggests()` with appropriate `skip_if_not_installed()` bcbb80dafedb023d53f249630328d0f3ee791f91
* Did minor test refactoring where I saw it useful bcbb80dafedb023d53f249630328d0f3ee791f91
* simplified some dplyr expressions in examples d2aa113762cefe65bc081e130c34f4a5f9ae4f6b, bc70cdb4b10310a594863a00c75890811d22caff

<details><summary>I verified that they are all correct with this code
</summary>
<p>

```r
v1 <-  towny |>
     dplyr::select(name, csd_type, census_div, population_2021) |>
     dplyr::group_by(census_div) |>
     dplyr::summarize(
       population = sum(population_2021),
       .groups = "drop_last"
     ) |>
     dplyr::arrange(population) |>
     dplyr::slice_head(n = 5) |>
     dplyr::mutate(ranking = dplyr::row_number()) |>
     dplyr::select(ranking, dplyr::everything()) 
v2 <- towny |>
  dplyr::select(name, csd_type, census_div, population_2021) |>
  dplyr::group_by(census_div) |>
  dplyr::summarize(
    population = sum(population_2021),
    .groups = "drop_last"
  ) |>
  dplyr::slice_min(population, n = 5) |>
  dplyr::mutate(ranking = dplyr::row_number(), .before = 0) 

waldo::compare(
  v1, v2
)
```

</p>
</details> 

Other considerations:

When I noticed it, avoid self-reference, and 2 cross-refs to the same function in the same paragrah.
The diff is large because I reflowed roxygen comments to be about 80 characters long.

### CODE REVIEW

**Comment by rich-iannone** [comment]
This is monumental work! I am super impressed so far (and it's okay that this touches so many files, it has to).

**Comment by olivroy** [comment]
Thanks for your interest! I pushed another chunk of changes to go around rough edges. I will do another self-review (probably tomorrow) and push the necessary fixes

**Review by rich-iannone — APPROVED** [review]
LGTM!
